src/packet.ml

```diff
@@ -430,8 +465,8 @@ module Tls_crypt = struct
           in
```

Comment:
in the line above, the `encrypted` includes the 2 byte length of wkc -- likely that should be discarded... the split should be `encrypted .. 2 byte wkc len .. wkc`` -- so the middle part (length) can be safely discarded

Reply:
The wkc len is part of the authenticated data so we need to preserve it.

Reply:
ok, but should it be part of `encrypted`? (not sure what is done with encrypted afterwards)

Reply:
No, wkc is not encrypted with the client key (though parts of it is encrypted with the server key). The `wkc_len` is the length of the wkc *including* the `wkc_len` btw.

Reply:
oh, interesting. I wasn't aware the length would include the length field itself. I suspect then there's nothing to do here :)